lib/pure/math.nim

```diff
@@ -980,27 +991,44 @@ func euclMod*[T: SomeNumber](x, y: T): T {.since: (1, 5, 1).} =
   if result < 0:
     result += abs(y)
 
-when not defined(js):
-  func c_frexp*(x: float32, exponent: var int32): float32 {.
-      importc: "frexp", header: "<math.h>".}
-  func c_frexp*(x: float64, exponent: var int32): float64 {.
-      importc: "frexp", header: "<math.h>".}
-  func frexp*[T, U](x: T, exponent: var U): T =
-    ## Split a number into mantissa and exponent.
-    ##
-    ## ``frexp`` calculates the mantissa m (a float greater than or equal to 0.5
-    ## and less than 1) and the integer value n such that ``x`` (the original
-    ## float value) equals ``m * 2**n``. frexp stores n in `exponent` and returns
-    ## m.
-    ##
-    runnableExamples:
-      var x: int
-      doAssert frexp(5.0, x) == 0.625
-      doAssert x == 3
-    var exp: int32
-    result = c_frexp(x, exp)
+func frexp*[T: float32|float64](x: T, exponent: var int): T =
+  ## Splits a number into mantissa and exponent.
+  ##
+  ## `frexp` calculates the mantissa m (a float greater than or equal to 0.5
+  ## and less than 1) and the integer value n such that `x` (the original
+  ## float value) equals `m * 2**n`. frexp stores n in `exponent` and returns
+  ## m.
+  ##
+  runnableExamples:
+    var x: int
+    doAssert frexp(5.0, x) == 0.625
+    doAssert x == 3
+
```

Comment:
the other way around, the new, idiomatic, preferred way should contain the implementation instead, and this can simply be:
```nim
  runnableExamples:
    var x: int
    doAssert frexp(5.0, x) == 0.625
    doAssert x == 3
  (result, exponent) = frexp(x)
```

Reply:
But new procs should be marked as `since(1, 5, 1)`.

Reply:
this should work:
```nim
func frexp[T: float32|float64](x: T, exponent: var int): T =
since (1,5,1):
  export frexp
```
in future work, there can be a more idiomatic `exportSince` (https://github.com/timotheecour/Nim/issues/479)